lsp/src/lsp-diagnostics.c

```diff
@@ -615,6 +636,9 @@ void lsp_diagnostics_show_all(void)
 
 		foreach_ptr_array(diag, i, diags)
 		{
+			if (current_doc_only && !utils_str_equal(doc->file_name, key))
```

Comment:
Better `doc->real_path` - I use this one everywhere.

(I've just noticed that I don't convert it to UTF-8 to display in the message window, will fix.)

Reply:
Ah, good that you noticed.
I intended to ask about this and forgot :).
Fixed.